templates/geosafe/analysis/dynamic_scripts/create/main.html

```diff
@@ -56,7 +56,7 @@
                         aggregation_name = value.feature.properties[aggregation_field_name];
                         (aggregation_name) ? aggregation_names.push(aggregation_name) : false;
                     });
-                    if (aggregation_names) {
+                    if (aggregation_names && aggregation_names.length > 0) {
```

Comment:
Only do filtering if `aggregation_names` has elements.
(If `features_collection['values']` is an empty list, that means the resulting aggregation layer is an empty layer. @myarjunar probably doesn't mean this.)